lib/smart_proxy_dynflow_core/callback.rb

```diff
@@ -33,13 +33,13 @@ def callback_resource
       end
 
       def ssl_options
-        return {} unless Settings.instance.use_https
-        client_key = File.read  Settings.instance.ssl_private_key
-        client_cert = File.read Settings.instance.ssl_certificate
+        return {} unless URI.parse(Settings.instance.callback_url).scheme == 'https'
```

Comment:
I don't think so, we need the use_https do decide, if we the core itself as http or https: it's not related directly on what to use for authenticating against callback